### PR TITLE
CID 1383925: Fix null check

### PR DESCRIPTION
**Pull request by masaori335**

### CODE REVIEW

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -5477,20 +5477,21 @@ HttpSM::handle_server_setup_error(int event, void *data)
...
+      // If we're writing the request and get an inactivity timeout
+      //   before any bytes are written, the connection to the
+      //   server failed
+      // In case of TIMEOUT, the iocore sends back
+      // server_entry->read_vio instead of the write_vio
+      // if (vio->op == VIO::WRITE && vio->ndone == 0) {
+      if (server_entry->write_vio && server_entry->write_vio->nbytes > 0 && server_entry->write_vio->ndone == 0) {
+        t_state.current.state = HttpTransact::CONNECTION_ERROR;
+      } else {
+        t_state.current.state = HttpTransact::INACTIVE_TIMEOUT;
```
Comment (bryancall) [inline proxy/http/HttpSM.cc]: We should set `t_state.current.state = HttpTransact::INACTIVE_TIMEOUT` if even if server_entry is nullptr

Comment (maskit) [reply proxy/http/HttpSM.cc]: In that case, is that really inactive timeout? It might be a connection error.

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by masaori335** [comment]
[approve ci autest]

**Comment by masaori335** [comment]
This build error on autest is fixed by #3659 ?
```
Log.cc: In static member function 'static void Log::init_fields()':
Log.cc:367:61: error: cast between incompatible function types from 'int (*)(char**, char*, int, LogSlice*)' to 'LogField::UnmarshalFunc' {aka 'int (*)(char**, char*, int)'} [-Werror=cast-function-type]
                        (LogField::UnmarshalFunc)&LogAccess::unmarshal_str);
                                                             ^~~~~~~~~~~~~
```

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Pull request merged**